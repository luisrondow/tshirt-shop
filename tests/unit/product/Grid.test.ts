import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'

import ProductGrid from '~/components/product/Grid.vue'

describe('ProductGrid component', () => {
  it('renders the product grid correctly', () => {
    const wrapper = mount(ProductGrid, {
      global: {
        plugins: [
          createTestingPinia({
            initialState: {
              products: {
                products: [
                  { id: '1', name: 'Product 1', price: 100, stock: 10, image: 'mock' },
                  { id: '2', name: 'Product 2', price: 200, stock: 20, image: 'mock' },
                ],
              },
            },
          }),
        ],
      },
    })

    expect(wrapper.text()).toContain('Product 1')
    expect(wrapper.text()).toContain('Product 2')

    expect(wrapper.text()).toContain('€100')
    expect(wrapper.text()).toContain('€200')

    expect(wrapper.element).toMatchSnapshot()
  })
})
