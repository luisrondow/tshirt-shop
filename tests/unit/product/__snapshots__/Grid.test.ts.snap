// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductGrid component > renders the product grid correctly 1`] = `
<div
  class="grid grid-cols-2 gap-x-2 gap-y-4 md:grid-cols-4 lg:gap-x-4 lg:gap-y-8"
>
  
  <div
    class="flex flex-col"
  >
    <img
      alt="product"
      class="h-auto w-full object-cover"
      src="mock"
    />
    <div
      class="flex items-center justify-between py-4"
    >
      <h1
        class="font-serif text-2xl font-semibold"
      >
        Product 1
      </h1>
      <p
        class="font-mono text-xl font-semibold"
      >
        €100
      </p>
    </div>
    <button
      class="h-10 w-full uppercase border border-black bg-black px-4 py-2 text-white  hover:bg-white hover:text-black"
    >
      
       Add to basket 
      
    </button>
  </div>
  <div
    class="flex flex-col"
  >
    <img
      alt="product"
      class="h-auto w-full object-cover"
      src="mock"
    />
    <div
      class="flex items-center justify-between py-4"
    >
      <h1
        class="font-serif text-2xl font-semibold"
      >
        Product 2
      </h1>
      <p
        class="font-mono text-xl font-semibold"
      >
        €200
      </p>
    </div>
    <button
      class="h-10 w-full uppercase border border-black bg-black px-4 py-2 text-white  hover:bg-white hover:text-black"
    >
      
       Add to basket 
      
    </button>
  </div>
  
</div>
`;
